src/ALE/coord_rho.F90

```diff
@@ -5,6 +5,7 @@ module coord_rho
 
 use MOM_error_handler, only : MOM_error, FATAL
 use MOM_remapping,     only : remapping_CS, remapping_core_h
+use MOM_unit_scaling,  only : unit_scale_type
```

Comment:
I actually thought about this issue a great deal while working up this contribution. There are actually two scaling arguments that would be needed, one to scale density and another to invert the scaling of pressure, but these are not separate arguments to calculate_density. (They were separate in intermediate commits within this PR, but the code was getting ugly and unwieldy.) Also, the unit_scale_type is already in use within MOM_EOS, so there are no new module dependencies being introduced by this change to `build_rho_column()`.  Moreover the unit_scale_type actually has _fewer_ MOM6 module dependencies than the hor_index_type, and it is already in use across MOM6 and SIS2, so I do not think that this is a problem.

  For `int_density_dz_wright()`, there was not a concern with accommodating a lower interface, so having two lengthy and potentially confusing arguments (one scales from pressure units to MKS units, while the other does the reverse) is less problematic than for arguments that would be called in places where there are more likely to be user contributions. I would be happy to discuss this further, or to make US into an optional argument.

Reply:
An optional argument does not remove the dependency on a MOM type so that is not a solution. We do need a broader discussion about how to apply dimensional tests to code that doesn't use MOM6 types but this case is quite clear - the new dependency can be avoided quite easily.

Reply:
I think that we have a philosophical disagreement here, as well as practical considerations. There are two variable units being rescaled, not one, and this may eventually become 4.  There is no variant of the equation of state with 2 or 4 optional naked rescaling variables. At one point I did have a variant with multiple scaling arguments, and it seemed much worse to me than having all of them packaged up together in a single simple type that introduces no new dependencies on MOM6 modules beyond what is already present due to the use of the equation of state.

Reply:
You added the scaling arguments to MOM_EOS_Wright easily enough but you chose to unpack US in MOM_EOS rather than above. That is what has led to the column-wise ALE functions needing to use US. I am saying this is a mistake. It severely impacts our plans to interface to those functions from non-MOM code.

Reply:
Here's how I'd solve this problem, in pseudo-pseudo-code:
```
type :: EOS_type
...
real: T_scaling, P_scaling ! etc
```
which mean US can be unpacked by MOM when initializing EOS and then calls to EOS no longer need US.

Reply:
I'm sending you a patch to implement this.

Reply:
I'm just reconciling conflicts between my patch and your revision...